packages/timeline-state-resolver/src/integrations/casparCG/index.ts

```diff
@@ -849,83 +699,70 @@ export class CasparCGDevice extends DeviceWithState<State, DeviceOptionsCasparCG
 		}
 		this.emitDebug(cwc)
 
-		return this._ccg
-			.do(cmd)
-			.then((resCommand) => {
-				if (this._queue[resCommand.token]) {
-					delete this._queue[resCommand.token]
-				}
-				// If the command was performed successfully, copy the state from the current state into the tracked caspar-state:
-				// This is later used in _assertIntendedState
-				if (
-					(resCommand.name === 'LoadbgCommand' ||
-						resCommand.name === 'PlayCommand' ||
-						resCommand.name === 'LoadCommand' ||
-						resCommand.name === 'ClearCommand' ||
-						resCommand.name === 'StopCommand' ||
-						resCommand.name === 'ResumeCommand') &&
-					resCommand.channel &&
-					resCommand.layer
-				) {
-					const currentState = this.getState(time)
-					if (currentState) {
-						const currentCasparState = currentState.state
-
-						const trackedState = this._ccgState.getState()
-
-						const channel = currentCasparState.channels[resCommand.channel]
-						if (channel) {
-							if (!trackedState.channels[resCommand.channel]) {
-								trackedState.channels[resCommand.channel] = {
-									channelNo: channel.channelNo,
-									fps: channel.fps || 0,
-									videoMode: channel.videoMode || null,
-									layers: {},
-								}
+		const { request, error } = await this._ccg.executeCommand(cmd)
+		if (error) {
+			this.emit('commandError', error, cwc)
+		}
+
+		try {
+			const response = await request
+
+			// I forgot what this means.. oh well... todo
```

Comment:
Would be really nice if we didn't have comments like this in the code...

Reply:
Probably in regards as to why the resolved result of `request` can be undefined. Depending on that returning may or may not be the correct thing to do here..